src/ray/object_manager/plasma/store.h

```diff
@@ -48,13 +49,98 @@ using flatbuf::PlasmaError;
 
 struct GetRequest;
 
+using CreateObjectCallback = std::function<Status(const std::shared_ptr<Client> &client, bool reply_on_oom, bool evict_if_full)>;
+
+class CreateRequestQueue {
```

Comment:
TODO: unit test for this class => will you handle it in this PR? I am a little afraid of merging the PR without unit testing this component.